### ISSUE TITLE
[BUG] : Environment Inject is not working in Init Containers

### ISSUE THREAD

**Issue by pvmraghunandan** [SparebankenVest/azure-key-vault-to-kubernetes#141]
**Description**

We are trying to inject key vault secrets to Init Container for doing few shell operations. Unfortunately, we couldn't make it work. Init container is always stuck. We tried checking environment injector logs and not much info or activity there.

**To Reproduce**
Below is the sample YAML File

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: hello-kubernetes
  namespace: com-test-dev
spec:
  replicas: 1
  selector:
    matchLabels:
      app: hello-kubernetes
  template:
    metadata:
      labels:
        app: hello-kubernetes
    spec:
      volumes:
        - name: sample
          emptyDir: {}
      initContainers:
      - name: init-test
        image: busybox
        command: ['sh', '-c', 'echo "$test" > "/opt/sample/test";']
        volumeMounts:
        - name: testvolume
          mountPath: /opt/sample
        env:
        - name: test
          value: "testsecret@azurekeyvault" # ref to akvs
      containers:
      - name: hello-kubernetes
        image: paulbouwer/hello-kubernetes:1.8
        ports:
        - containerPort: 8080
        volumeMounts:
        - name: sample
          mountPath: /opt/sample
```

**Expected behavior**
Init Container should have secrets injected.

**Comment by pvmraghunandan** [comment]
Looks like it's not working with busybox but i am not completely sure though

**Comment by torresdal** [comment]
@pvmraghunandan thanks for reporting this. You say the init-container is stuck - what does that mean? Are there anything in the init-container logs? Any errors?

**Comment by pvmraghunandan** [comment]
@torresdal No errors and no logs when we tried checking. kubectl describe pod shows that init container is in Running state forever.

**Comment by torresdal** [comment]
@pvmraghunandan What if the env-variable is just plain text, and not a env-injected one?

**Comment by pvmraghunandan** [comment]
@torresdal  that works perfectly fine. Also, if we remove busybox image in init container and use some public image, it is working fine. We just validated today.

**Comment by pvmraghunandan** [comment]
@torresdal did you get a chance to check this? Somehow, it's not working with any init container

**Comment by torresdal** [comment]
A believe it's related to a container inspection bug we have in 1.1 and not to do with init-container - could that be? In that case it is solved in 1.2-beta, soon to be released.

**Comment by 181192** [comment]
@pvmraghunandan Did you get a change to test the 1.2 release of akv2k8s? Hopefully this should be fixed now. 
Closing this issue for now, but feel free to re-open if this is still an issue for you :blush:
